stl/src/cthread.cpp

```diff
@@ -103,9 +103,7 @@ _CRTIMP2_PURE _Thrd_id_t __cdecl _Thrd_id() noexcept { // return unique id for c
 }
 
 _CRTIMP2_PURE unsigned int __cdecl _Thrd_hardware_concurrency() noexcept { // return number of processors
-    SYSTEM_INFO info;
-    GetNativeSystemInfo(&info);
-    return info.dwNumberOfProcessors;
+    return GetActiveProcessorCount(ALL_PROCESSOR_GROUPS);

```

Comment:
Unfortunately this won't work because [`GetActiveProcessorCount`](https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getactiveprocessorcount) is "Windows 7 \[desktop apps only\]" and the STL also needs to support UWP apps.

We'll need to follow Raymond Chen's "hard way" and call [`Get­Logical­Processor­Information­Ex`](https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getlogicalprocessorinformationex) which is "Windows 7 \[desktop apps | UWP apps\]" and therefore always available to use.